src/components/Clock.tsx

```diff
@@ -0,0 +1,61 @@
+import { Component } from 'react';
+
+type State = {
+  today: string,
+};
+
+type Props = {
+  clockName: string,
+};
+
+function updateTime(): string {
+  return (new Date()).toUTCString().slice(-12, -4);
+}
+
+export class Clock extends Component<Props, State> {
+  state: Readonly<State> = {
+    today: updateTime(),
+  };
+
+  clockTimerId = 0;
+
+  componentDidMount() {
+    this.clockTimerId = window.setInterval(() => {
+      this.setState({ today: updateTime() });
+      // eslint-disable-next-line no-console
+      console.info(this.state.today);
+    }, 1000);
+  }
+
+  componentDidUpdate(prevProps: Props) {
+    if (prevProps.clockName !== this.props.clockName) {
+      // eslint-disable-next-line no-console
+      console.debug(
+        `Renamed from ${prevProps.clockName} to ${this.props.clockName}`,
+      );
+    }
+  }
+
+  componentWillUnmount() {
+    window.clearInterval(this.clockTimerId);
+  }
+
+  render() {
+    const { today } = this.state;
+    const { clockName } = this.props;
+
+    return (
+      <div className="Clock">
+        <strong className="Clock__name">
+          {clockName}
+        </strong>
+
+        {' time is '}
+
+        <span className="Clock__time">
+          {today}
+        </span>
```

Comment:
What is this span for if the class is not used?
```suggestion
        {`  time is  ${today}`}
```

Reply:
This span with className="Clock__time" is needed in tests. Without this, tests fail.
![Знімок екрана 2023-01-28 о 11 36 21](https://user-images.githubusercontent.com/112945779/215258982-7bfa7e1a-3b6b-4bb0-8c04-f1905e6e5d65.png)